packages/aws-cdk/lib/api/hotswap/cloudwatch-logs.ts

```diff
@@ -0,0 +1,97 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { LazyListStackResources } from '../hotswap-deployments';
+import { Mode, SdkProvider } from './../aws-auth';
+import { EvaluateCloudFormationTemplate } from './evaluate-cloudformation-template';
+import { CloudWatchLogEventMonitor } from './monitor/logs-monitor';
+
+// resource types that have associated CloudWatch Log Groups that should
+// be monitored
+const includedTypes = ['AWS::StepFunctions::StateMachine', 'AWS::ECS::TaskDefinition'];
+
+// resource types that will create a cloudwatch log group
+// with a specific name if one is not provided
+const implicitTypes = ['AWS::Lambda::Function', 'AWS::CodeBuild::Project'];
+
+export async function registerCloudWatchLogGroups(
+  sdkProvider: SdkProvider,
+  stackArtifact: cxapi.CloudFormationStackArtifact,
+  assetParams: { [key: string]: string },
+  hotswapLogMonitor: CloudWatchLogEventMonitor,
+): Promise<void> {
+
+  const logGroupNames = new Array<string>();
+  const logGroupLogicalIds = new Set<string>();
+  const implicitMap = new Map<string, string>();
+  //
+  // resolve the environment, so we can substitute things like AWS::Region in CFN expressions
+  const resolvedEnv = await sdkProvider.resolveEnvironment(stackArtifact.environment);
+  // create a new SDK using the CLI credentials, because the default one will not work for new-style synthesis -
+  // it assumes the bootstrap deploy Role, which doesn't have permissions to update Lambda functions
+  const sdk = await sdkProvider.forEnvironment(resolvedEnv, Mode.ForWriting);
+
+  // add the sdk to the hotswapLogMonitor so that it also has the
+  // correct credentials to monitor CloudWatch log groups
+  hotswapLogMonitor.addSdk(sdk);
+  // The current resources of the Stack.
+  // We need them to figure out the physical name of a resource in case it wasn't specified by the user.
+  // We fetch it lazily, to save a service call, in case all hotswapped resources have their physical names set.
+  const listStackResources = new LazyListStackResources(sdk, stackArtifact.stackName);
+  const evaluateCfnTemplate = new EvaluateCloudFormationTemplate({
+    stackArtifact,
+    parameters: assetParams,
+    account: resolvedEnv.account,
+    region: resolvedEnv.region,
+    partition: (await sdk.currentAccount()).partition,
+    urlSuffix: sdk.getEndpointSuffix,
+    listStackResources,
+  });
```

Comment:
If you need to move this classes out of the `hotswap` directory, because they will be needed in other places, I'm not opposed to that.

Reply:
I moved these up a level.